### ISSUE TITLE
Mutually exclusive vs. last one wins

### ISSUE THREAD

**Issue by nicolasr75** [commandlineparser/commandline#381]
Assume I have some mutually exclusive boolean options like
--device-time
--local-time
--utc

A customer had the following request: 
he has a rather long list of commandline options for my program stored in a batch file. One of them might be --device-time. At times he would like to override it by appending another option like --local-time. That is, the options would not be mutually exclusive anymore but the last one wins.

Since the library of course doesn't know about this I would have to implement this myself. The question is how? I would need to know the order in which the options were specified. Is something like this possible?

**Comment by ericnewton76** [comment]
I believe this might be default behavior.  

Your Options class should probably detect this situation and Log a warning of which value is actually in effect.

**Comment by nicolasr75** [comment]
It seems to me that all I can detect from the ParserResult is, that both flags are set. Without knowing the order of specification I can neither implement "last one wins" nor can I log a warning of which value is actually in effect. Or did I miss something?

**Comment by ericnewton76** [comment]
I believe you're correct.  (Sorry I'm not 100% familiar with the internals without looking)

This certainly looks like a good scenario that needs to be addressed... 

I'm seeing a few issues being registered here that can be solved with a similar solution of having the ParserResult include a list of the parsed options and values (and potentially how each individual item was handled... error, type conversion failure/success, or otherwise

**Comment by nicolasr75** [comment]
Yes, that sounds good. I haven't looked at the code myself but I imagine something like a dictionary instead of a list, keyed by the option and giving the value an object with the details as you say, including the index. That way I could do something like:

```
if(whatever[--local-time].Index > whatever[--device-time].Index)
   //--local-time wins
```
The application could then or course decide whether the first or the last option wins, or reject it.
Just a sponanteous idea though.

**Comment by ericnewton76** [comment]
Perhaps theres a more generic way, being able to specify that both options are mutually exclusive.  I believe theres a way in the Option attribute that will influence the parser

Realistically I think there should be a little more validation possibility baked into the parsing process.

**Comment by nicolasr75** [comment]
Yes, but let me rephrase my scenario: not the specification of the options is mutually exclusive but only the application. If multiple options of the same group are specified, this is resolved by last-one-wins or whatever custom logic. I think there is not even a need for an actual group. I just need to be able to inject my custom code somehow and be able to manipulate the parser result.

Actual Mutual Exclusion to me means that the parser itself generates an error when you specify two options of the same group and that is a good feature to have!

**Comment by moh-hassan** [comment]
use three different setname as below:
```cs
class Options
{
	[Option("device-time",SetName="device",Required=true)]
	public DateTime DeviceTime {get;set;}
	[Option("local-time",SetName="local",Required=true)]
	public DateTime LocalTime {get;set;}
	[Option("utc",SetName="utc",Required=true)]
	public DateTime Utc {get;set;}
}

```
[more examples](https://github.com/commandlineparser/commandline/wiki/Mutually-Exclusive-Options)

**Comment by nicolasr75** [comment]
Hmm, Sets are definitely a great feature but I don't understand how that solves my problem??? As I emphasized my user wants to be able to overwrite an option occasionally. That rules out mutually exlusivity as provided by CommandLineParser. I have to write my own code that manipulates the parser result to make "last one wins" work but for that I need the order of the passed options.

**Comment by moh-hassan** [comment]
>my user wants to be able to overwrite an option occasionally.

The "last one wins" is  a new feature in the next v2.9. you can try it using the package in [CI server](https://ci.appveyor.com/api/buildjobs/uqhxtrgogmde69g7/artifacts/src%2FCommandLine%2Fbin%2FRelease%2FCommandLineParser.2.8.0-ci-140.nupkg), and set parser as:
```cs
var parser= new Parser(settings => settings.AllowMultiInstance = true);
```
Just, to coorect:
>Actual Mutual Exclusion to me means that the parser itself generates an error when you specify two options of the same group

Actual Mutual Exclusive means that the parser itself generates an error when you specify options in two or more different SetName,  i.e only one set at a time is allowed, plus the common options that is not included in a Set.
